service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerTest.java

```diff
@@ -0,0 +1,235 @@
+import uk.gov.hmcts.reform.fpl.service.UploadDocumentService;
+
+import java.util.List;
+import java.util.UUID;
+
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.ALL_PARTIES;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.CAFCASS;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.COURT;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.LOCAL_AUTHORITY;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.OTHERS;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.PARENTS_AND_RESPONDENTS;
+import static uk.gov.hmcts.reform.fpl.utils.DocumentManagementStoreLoader.document;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(DraftOrdersController.class)
+@OverrideAutoConfiguration(enabled = true)
+class DraftOrdersControllerTest {
+
+    @MockBean
+    private DocmosisDocumentGeneratorService documentGeneratorService;
+
+    @MockBean
+    private UploadDocumentService uploadDocumentService;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String USER_ID = "1";
+
+    @Test
+    void aboutToStartCallbackShouldSplitDirectionsIntoSeparateCollections() throws Exception {
+        String title = "example direction";
+
+        List<Direction> directions = ImmutableList.of(
+            Direction.builder().type(title).assignee(ALL_PARTIES).build(),
+            Direction.builder().type(title).assignee(LOCAL_AUTHORITY).build(),
+            Direction.builder().type(title).assignee(PARENTS_AND_RESPONDENTS).build(),
+            Direction.builder().type(title).assignee(CAFCASS).build(),
+            Direction.builder().type(title).assignee(OTHERS).build(),
+            Direction.builder().type(title).assignee(COURT).build()
+        );
+
+        Order sdo = Order.builder().directions(buildDirections(directions)).build();
+
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .data(ImmutableMap.of("standardDirectionOrder", sdo))
+                .build())
+            .build();
+
+        MvcResult response = mockMvc
+            .perform(post("/callback/draft-SDO/about-to-start")
+                .header("authorization", AUTH_TOKEN)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(mapper.writeValueAsString(request)))
+            .andExpect(status().isOk())
+            .andReturn();
+
```

Comment:
Maybe we could extract common parts, just to make this a bit smaller.

Reply:
sure, extracted to private makeRequest method 👍